tests/foreman/api/test_syncplan.py

```diff
@@ -639,6 +639,10 @@ def test_negative_synchronize_custom_product_past_sync_date(module_org):
     with pytest.raises(AssertionError):
         validate_task_status(repo.id, max_tries=2)
     validate_repo_content(repo, ['erratum', 'package', 'package_group'], after_sync=False)
+    # Disable sync plan after the test to reduce load on Satellite
+    sync_plan.enabled = False
+    sync_plan = sync_plan.update(['enabled'])
+    assert sync_plan.enabled is False
```

Comment:
Stephen this LGTM.
one query, should we create a separate sync_plan_disable function(teardown) to handle this?

Reply:
I don't know what benefit that would have, e.g. would it be faster or easier to follow? It not really a teardown, so I reluctant to change it.

Reply:
@swadeley I agree with @jameerpathan111 you have used that 4 line of code to disable the sync plan which is not part of test and used multiple times. It  can be taken as teardown or fixture that take care of disabling the sync plan

Reply:
@jameerpathan111 @omkarkhatavkar I would agree, its better to not repeat this block so many times. I would definitely consider it cleanup/teardown, because its not related to the test coverage, and should be part of the pytest TEARDOWN phase.  With it in the CALL phase like this, an assert would look like a test failure, not a failure in disabling the sync_plan after the test coverage is complete.  One additional consideration is that the test may fail before these statements are reached, and the sync_plan will be left on the system, enabled.

Given the sync_plans are created within the tests here, I think you should define a function that includes these three statements, and takes `sync_plan` as the argument.

These test functions should call `request.addfinalizer()` after creating the sync plan, so that it gets disabled at the proper scope, after the test call phase finishes.

Reply:
I have added request.addfinalizer step to API, CLI, and UI modules.